templates/model/sdk_update_pre_build_request.go.tpl

```diff
@@ -0,0 +1,5 @@
+// this to handle delete/remove tags 
+_ , err = rm.deleteTags(ctx,desired,latest)
+if err != nil {
+	return nil, err
+}
```

Comment:
We don't need to delete the tags before performing a tag update.

Reply:
@a-hilaly  - as I mentioned above , SDK update method is handling ADDTAGS only  and hooks are deleting tags , there is no modify API available for sagemaker model resource . So in SDKUPDATE , first "DeletTags" method look for any deletion request , if not then proceed and look for add tags request .

Let me know if anything else is require.